### PR TITLE
Add custom vision

### PR DESCRIPTION
**Pull request by nicolasbotto**

### CODE REVIEW

**Comment by avkewalr-msft** [comment]
looks good to me. #sign-off @naiteeks - take one look please

File: edge-modules/extensions/customvision/avaextension/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    && apt-get update && apt-get install -y --no-install-recommends \
+    wget runit nginx \
+    && rm -rf /var/lib/apt/lists/* \
+    && apt-get clean \
+    && rm /etc/nginx/sites-enabled/default \
+    && cp /avaExtension/nginx/app /etc/nginx/sites-available/ \
+    && ln -s /etc/nginx/sites-available/app /etc/nginx/sites-enabled/
+
+EXPOSE 80
+CMD ["supervisord", "-c", "/avaExtension/etc/supervisord.conf"]
```
Comment (naiteeks) [inline edge-modules/extensions/customvision/avaextension/Dockerfile:27]: avaextension

File: edge-modules/extensions/customvision/avaextension/object_detection.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+        self.labels = labels
+        self.prob_threshold = prob_threshold
+        self.max_detections = max_detections
+
+    def _logistic(self, x):
+        return np.where(x > 0, 1 / (1 + np.exp(-x)), np.exp(x) / (1 + np.exp(x)))
+
+    def _non_maximum_suppression(self, boxes, class_probs, max_detections):
+        """Remove overlapping bouding boxes
```
Comment (naiteeks) [inline edge-modules/extensions/customvision/avaextension/object_detection.py:37]: Are there extra quotes ?

File: edge-modules/extensions/customvision/avaextension/object_detection.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        ratio = math.sqrt(self.DEFAULT_INPUT_SIZE / image.width / image.height)
+        new_width = int(image.width * ratio)
+        new_height = int(image.height * ratio)
+        new_width = 32 * round(new_width / 32);
+        new_height = 32 * round(new_height / 32);
+        image = image.resize((new_width, new_height))
+        return image
+
+    def predict(self, preprocessed_inputs):
+        """Evaluate the model and get the output
```
Comment (naiteeks) [inline edge-modules/extensions/customvision/avaextension/object_detection.py:140]: Are there Extra quotes?

**Review by naiteeks — COMMENTED** [review]
One fix needed snd extra quotes check

File: edge-modules/extensions/customvision/avaextension/Dockerfile (hunk 1 above)
Comment (avkewalr-msft) [reply edge-modules/extensions/customvision/avaextension/Dockerfile:27]: this is internal directory, should be ok.

File: edge-modules/extensions/customvision/avaextension/object_detection.py (hunk 2 above)
Comment (avkewalr-msft) [reply edge-modules/extensions/customvision/avaextension/object_detection.py:37]: should be fine

File: edge-modules/extensions/customvision/avaextension/object_detection.py (hunk 3 above)
Comment (avkewalr-msft) [reply edge-modules/extensions/customvision/avaextension/object_detection.py:140]: should be fine

**Comment by avkewalr-msft** [comment]
@gadamilan - can you please merge this

**Pull request merged**